src/viking.js

```diff
@@ -1,11 +1,116 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength){
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack(){
+        return this.strength;
+    }
+
+    receiveDamage(damage){
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength)
+        this.name = name;
+    }
+    receiveDamage(damage){
+        this.health -= damage;
+
+        if(this.health > 0) return `${this.name} has received ${damage} points of damage`;
+        else return `${this.name} has died in act of combat`
+    }
+
+    battleCry(){
+        return 'Odin Owns You All!'
+    }
+    
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength){
+        super(health, strength)
+        
+       
+
+    }
+
+    receiveDamage(damage){
+        this.health -= damage;
+        if(this.health > 0) return `A Saxon has received ${damage} points of damage`;
+        else return `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+
+        this.vikingArmy = [];
+        this.saxonArmy =[]; 
+
+    }
+
+    addViking(viking){
+
+        this.vikingArmy.push(viking);
+
+    }
+
+    addSaxon(saxon){
+
+        this.saxonArmy.push(saxon);
+
+    }
+
+    showStatus(){
+        if(this.saxonArmy.length <= 0) return 'Vikings have won the war of the century!';
+        if(this.vikingArmy.length <= 0) return 'Saxons have fought for their lives and survived another day...';
+        if(this.saxonArmy.length >= 1 && this.vikingArmy.length >= 1) return 'Vikings and Saxons are still in the thick of battle.'
+    }
+
+    vikingAttack(){
+
+        let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+        let randomSaxon =  Math.floor(Math.random() * this.saxonArmy.length);
+
+
+        let attackResult = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
+
+        if(this.saxonArmy[randomSaxon].health <= 0) this.saxonArmy.pop(randomSaxon);
```

Comment:
The array method `.pop()` will exclude only the last index of the array. Since the Saxon will always be random, and we don't know if it will be the first or the last index of the array, the correct method to use here is `.splice()` because it let's you exclude an element from any index of the array. The method `.splice()` accepts two arguments: the index from where you want to start to delete elements, and how many elements you want to delete:

```
(...)

    if (this.saxonArmy[randomSaxon].health <= 0)
      this.saxonArmy.splice(randomSaxon, 1);

(...)
```